go/mcap/reader.go

```diff
@@ -164,7 +179,38 @@ func (r *Reader) Info() (*Info, error) {
 		Schemas:           it.schemas,
 		Footer:            it.footer,
 		Header:            r.header,
-	}, nil
+	}
+	r.info = info
+	return info, nil
+}
+
+func (r *Reader) GetMetadata(offset uint64) (*Metadata, error) {
+	info, err := r.Info()
+	if err != nil {
+		return nil, err
+	}
+	for _, idx := range info.MetadataIndexes {
```

Comment:
What's the point of checking the MetadataIndexes if the offset is already known? This just seems like it would prevent reading metadata if there are no indexes?

Reply:
I suppose that's true. I guess I figured the only time anyone would be doing a read of a metadata record at a known offset would be through the index, but I could just blindly seek and try to parse and maybe that's more useful to someone.